client/Components/Form/ConfirmButton.jsx

```diff
@@ -14,23 +14,29 @@ import { Button } from 'react-bootstrap';
  */
 const ConfirmButton = ({ children, onClick }) => {
     const [showConfirm, setShowConfirm] = useState(false);
+    const [btnText, setBtnText] = useState(children);
 
     const handleClick = (event) => {
         event.preventDefault();
         setShowConfirm(!showConfirm);
+        if (btnText === 'Cancel') {
+            setBtnText(children);
+        } else {
+            setBtnText('Cancel');
```

Comment:
You'll need to pull in a translation object and pass this to it, so it's translated

```
const { t } = useTranslation();

setBtnText(t('Cancel'));
```

Reply:
I couldn't get this to work, but the translation object is already imported for the confirm button, so I did the following:

```jsx
if (showConfirm) {
    setBtnText(children);
} else {
    setBtnText(<Trans>Cancel</Trans>);
}
```

Which seems to work properly. Is that better?